frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js

```diff
@@ -131,16 +132,21 @@ const useCreateBLIsAndSCs = (
                                         redirectUrl: "/error"
                                     });
                                 });
+                        } else {
+                            // If no changes, return a resolved promise
+                            return Promise.resolve();
                         }
                     });
-                    resetForm();
-                    setIsEditMode(false);
-                    setAlert({
-                        type: "success",
-                        heading: "Agreement Edits Sent to Approval",
-                        message:
-                            "Your edits have been successfully sent to your Division Director to review. After edits are approved, they will update on the Agreement",
-                        redirectUrl: `/agreements/${selectedAgreement?.id}`
+                    Promise.all(promises).then(() => {
```

Comment:
You may want to also consider if one of the promises are rejected here

Reply:
https://github.com/HHS/OPRE-OPS/assets/4629398/8aea82bc-b137-41f2-84bb-752b8cc109dc

Reply:
solved with the Promise.allSettled

```js
 Promise.allSettled(promises).then((results) => {
                        let rejected = results.filter((result) => result.status === "rejected");
                        if (rejected.length > 0) {
                            console.error(rejected[0].reason);
                            setAlert({
                                type: "error",
                                heading: "Error Sending Agreement Edits",
                                message: "There was an error sending your edits for approval. Please try again.",
                                redirectUrl: "/error"
                            });
                        } else {
                            resetForm();
                            setIsEditMode(false);
                            setAlert({
                                type: "success",
                                heading: "Agreement Edits Sent to Approval",
                                message:
                                    "Your edits have been successfully sent to your Division Director to review. After edits are approved, they will update on the Agreement",
                                redirectUrl: `/agreements/${selectedAgreement?.id}`
                            });
                        }
```